tests/unified/runner.py

```
@@ -66,6 +68,86 @@ class TestExecutionError(Exception):
     pass
 
 
+async def send_discord_message(webhook_url: str, message: str) -> None:
+    """Send a message to Discord via webhook."""
+    try:
+        async with httpx.AsyncClient() as client:
+            response = await client.post(webhook_url, json={"content": message})
+            response.raise_for_status()
+            logger.info("Discord notification sent successfully")
+    except Exception as e:
+        logger.error(f"Failed to send Discord notification: {e}", exc_info=True)
```

Ruff (0.14.8)

Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)